application-licensing-licensor/application-licensing-licensor-api/src/main/java/com/xwiki/licensing/internal/upgrades/notifications/ExtensionAutoUpgradedEventDisplayer.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.licensing.internal.upgrades.notifications;
+
+import java.util.Arrays;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.LocalDocumentReference;
+import org.xwiki.notifications.CompositeEvent;
+import org.xwiki.notifications.NotificationException;
+import org.xwiki.notifications.notifiers.NotificationDisplayer;
+import org.xwiki.rendering.block.Block;
+import org.xwiki.template.Template;
+import org.xwiki.template.TemplateManager;
+
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+/**
+ * Displaying a custom template for ExtensionAutoUpgradedEvent. The NotificationDisplayerClass XObject should work
+ * independently of this class but it is unstable on older versions and sometimes the default template is rendered.
+ * After upgrading the 9.11 xwiki parent only the object should be used and this class removed.
+ * 
+ * @version $Id$
+ * @since 1.17
+ */
+@Component
+@Singleton
+@Named(ExtensionAutoUpgradedEventDisplayer.EVENT_TYPE)
+public class ExtensionAutoUpgradedEventDisplayer implements NotificationDisplayer
+{
+    /**
+     * The event type.
+     */
+    public static final String EVENT_TYPE =
+        "com.xwiki.licensing.internal.upgrades.notifications.ExtensionAutoUpgradedEvent";
```

Comment:
Isn't it safer to make this dynamic so that it doesn't get broken if we move the ``ExtensionAutoUpgradedEvent`` to a different package (for instance). As in:

    ExtensionAutoUpgradedEvent.class.getCanonicalName()